File: 2.coding-level-one/6.randomness/java/4.CoinGameSimulation.java
import java.util.Random;

public class CoinGameSimulation {

	// game of betting: how many heads will be turned up in 10 coin tosses?
	public static int getNumberOfHeads() {
		int nheads = 0;
		Random r = new Random();
		for (int i = 0; i < 10; ++i) {
			if (r.nextInt(2) == 0)
				nheads += 1;
		}
		return nheads;
	}

	public static void main(String[] args) {
		for (int i = 0; i < 10; ++i)
			System.out.println(getNumberOfHeads());
	}

}

File: 2.coding-level-one/6.randomness/cpp/4.coin game simulation.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

//game of betting: how many heads will be turned up in 10 coin tosses?
int getNumberOfHeads() {
	int nheads = 0;
	for (int i = 0; i < 10; ++i) {
		if (rand() % 2 == 0)
			nheads += 1;
	}
	return nheads;
}

/*int main() {
	for (int i = 0; i < 10; ++i)
		cout << getNumberOfHeads() << endl;
}*/
